delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Projects.scala

```diff
@@ -159,6 +165,44 @@ trait Projects {
     * @param offset the last seen event offset; it will not be emitted by the stream
     */
   def currentEvents(offset: Offset = NoOffset): Stream[Task, Envelope[ProjectEvent]]
+
+  /**
+    * Create a project for a user according to automatic provisioning settings.
+    *
+    * @param subject  the user for which to provision the project.
+    */
+  def provisionProject(subject: Subject): UIO[Unit]
+
+  private[delta] def provisionIfNotExists(
+      exists: Boolean,
+      projectRef: ProjectRef,
+      user: User,
+      acls: Acls,
+      provisioningConfig: AutomaticProvisioningConfig
+  ): IO[Unit, Unit] =
+    if (exists) UIO.unit
+    else {
+      val acl = Acl(AclAddress.Project(projectRef), user -> provisioningConfig.permissions)
+      (for {
+        _ <- acls
+               .append(acl, 0L)(user)
+               .onErrorRecover { case _: AclRejection.IncorrectRev => () }
```

Comment:
so if it returns IncorrectRev we are fine with it?

Reply:
It means that the Acls have already been set before for that project and we don't want to override those.

Reply:
Aren't the ACLs set for a project anyway once the project is created through `ScopeInitialization (OwnerPermissionsScopeInitialization)`

Reply:
But I think for this case we might want to have different permissions.

Reply:
I'm not sure it is worth the complexity and non deterministic behavior, since If there are already ACLs (IncorrectRev) we don't seem to mind _which_ permissions there are.

I'd rely on `OwnerPermissionsScopeInitialization` to do its job.

Reply:
There is a requirement for custom permissions in the issue https://github.com/BlueBrain/nexus/issues/2520, please discuss there if you feel that it's not needed https://github.com/BlueBrain/nexus/issues/2520